docs/events/glides/com.mbta.ctd.glides.vehicle_trip_assignments.v1.mdx

```diff
@@ -0,0 +1,184 @@
+
+### Vehicle
+
+Fields:
+
+- `vehicleId` (string): The vehicle ID that Glides gets from RTR via GTFS-RT. E.g. `"G-10223"`
+- `carNumbers` (array of strings | null): The list of human-readable car numbers of the train, in order from front to back. In GTFS-RT, this corresponds to the "label". E.g. `["3698", "3866"]`.
+
+The `vehicleId` should be used for checking for equality and tracking updates. The `carNumbers` MAY change (for example reversing the order, or updating bad data from a bad AVI read) while referring to the same train. Car numbers are included for context and to help match to other systems, but are not guaranteed to exactly match any other data source. `carNumbers` MAY be `null` if the vehicle has been removed from realtime tracking and there are no car numbers associated with it anymore.
+
+Glides SHOULD NOT publish a sequence of events that imply the same car number is associated with two different `vehicleId`s at the same time.
+
+### VehicleTripAssignment
+
+Fields:
+
+- `vehicle` ([Vehicle](#vehicle) | null):
+- `tripKey` ([TripKey](#tripkey) | null):
+
+`null` means "unassigned":
+
+- An assignment that includes both a `tripKey` and a `vehicle` means that the given vehicle is operating the given trip at this point in time.
+- An assignment that includes a `tripKey` but a `null` `vehicle` means that no known vehicle is operating that trip right now.
+- An assignment that includes a `vehicle` but a `null` `tripKey` means that the vehicle is not currently operating a trip.
+- Assignments SHOULD NOT have both fields `null`, but if they do, they can be ignored.
+
+## Updating Assignments
+
+Consumers SHOULD maintain state about which vehicle-trip assignments are active, and update that state when they receive an assignment that refers to a `vehicleId` or `tripKey`.
+
+Vehicles can only be assigned to one trip at a time. A vehicle that was previously assigned to `TRIP A` MAY be assigned directly to `TRIP B` without a separate assignment that assigns its trip to `null` first. Conversely, a trip can only have one assigned vehicle at a time, and doesn't need a previous vehicle to be removed before having a new vehicle assigned to it.
+
+If `VEHICLE 1` was last assigned to `TRIP A`, and a new event assigns `VEHICLE 1` to `TRIP B`, Glides MUST also send another `VehicleTripAssignment` in the same event's batch to assign the `vehicle` for `TRIP A` to `null` or to a different vehicle. Consumers are not expected to deduce that `TRIP A` no longer has a vehicle assigned based on an update to `VEHICLE 1` and `TRIP B` (which doesn't mention `TRIP A`). Glides MUST ensure that after processing all updates in an event's batch, if the last update to `TRIP A` assigned `VEHICLE 1` to it, then `VEHICLE 1` MUST NOT have any other more recent updates.
```

Comment:
**question:**

> Consumers are not expected to deduce that `TRIP A` no longer has a vehicle assigned based on an update to `VEHICLE 1` and `TRIP B` (which doesn't mention `TRIP A`).

Why don't we want to place this expectation on clients? It seems like straightforward logic to implement, and this seems to add complications to the schema itself. In particular, I don't love that it means that it matters whether two assignment events are in the same batch event as opposed to coming immediately one after the other in separate batches.

Reply:
I ended up actually making a pretty similar point in [your thread about the "alternative approach"](https://github.com/mbta/schemas/pull/9#discussion_r1844127293), so it might make sense to just put your combined response there.

Reply:
Resolved in [00399b4](https://github.com/mbta/schemas/pull/9/commits/00399b406442269fc470c6f9e1aa0fe3e96f3c85)